Essentials/src/main/java/com/earth2me/essentials/EssentialsUpgrade.java

```diff
@@ -146,6 +147,45 @@ public static void uuidFileConvert(final IEssentials ess, final Boolean ignoreUF
         ess.getLogger().info("To rerun the conversion type /essentials uuidconvert");
     }
 
+    public void convertMailList() {
+        if (doneFile.getBoolean("updateUsersMailList", false)) {
+            return;
+        }
+
+        final File userdataFolder = new File(ess.getDataFolder(), "userdata");
+        if (!userdataFolder.exists() || !userdataFolder.isDirectory()) {
+            return;
+        }
+        final File[] userFiles = userdataFolder.listFiles();
+        for (File file : userFiles) {
+            if (!file.isFile() || !file.getName().endsWith(".yml")) {
+                continue;
+            }
+            final EssentialsConfiguration config = new EssentialsConfiguration(file);
+            try {
+                config.load();
+                if (config.hasProperty("mail") && config.isList("mail")) {
+                    final ArrayList<MailMessage> messages = new ArrayList<>();
+                    for (String mailStr : Collections.synchronizedList(config.getList("mail", String.class))) {
+                        if (mailStr == null) {
+                            continue;
+                        }
+                        messages.add(new MailMessage(false, true, null, null, 0L, 0L, mailStr));
+                    }
+                    config.removeProperty("mail");
+                    config.setProperty("mail", messages);
```

Comment:
you need to pass the element type of the list through to configurate here somehow -- `.set(Object)` doesn't properly unwrap included values

could be `.set(TypeToken<List<MailMessage>>, messages)`, or `.setList(MailMessage.class, messages)`